### PR TITLE
[CARBONDATA-2071] Added block size to BblockletDataMap while initialising

### PR DESCRIPTION
**Pull request by ravipesala**
Added block size to blocklet datamap so that features like small file merge can use them

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [X] Any interfaces changed?
 
 - [X] Any backward compatibility impacted?
 
 - [X] Document update required?

 - [X] Testing done
             
 - [X] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3065/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1833/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3069/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1838/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3054/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3059/

**Comment by QiangCai** [comment]
verified on local standalone mode

LGTM

File: core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java [hunk 1]
```diff
@@ -78,17 +78,19 @@ public BlockletDataMap get(TableBlockIndexUniqueIdentifier identifier)
...
         CarbonFile carbonFile = FileFactory.getCarbonFile(segmentPath);
         CarbonFile[] carbonFiles = carbonFile.locationAwareListFiles();
         SegmentIndexFileStore indexFileStore = new SegmentIndexFileStore();
         indexFileStore.readAllIIndexOfSegment(carbonFiles);
         PartitionMapFileStore partitionFileStore = new PartitionMapFileStore();
         partitionFileStore.readAllPartitionsOfSegment(carbonFiles, segmentPath);
         for (CarbonFile file : carbonFiles) {
-          locationMap.put(file.getAbsolutePath(), file.getLocations());
+          blockMetaInfoMap
+              .put(file.getAbsolutePath(), new BlockMetaInfo(file.getLocations(), file.getSize()));
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java]: move `.put` to previous line

File: core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java [hunk 2]
```diff
@@ -244,4 +247,22 @@ public void clearAccessCount(
       cacheable.clear();
     }
   }
+
+  public static class BlockMetaInfo {
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java]: Is it only used in this class? I think it is better to move to outside if not just used in this class

File: core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java [hunk 3]
```diff
@@ -244,4 +247,22 @@ public void clearAccessCount(
...
+    private String[] locationInfo;
+    private long size;
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java]: please add comment for these two variable

File: core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java (hunk 2 above)
Comment (ravipesala) [reply core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java]: ok

File: core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java (hunk 3 above)
Comment (ravipesala) [reply core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java]: ok

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3074/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3064/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1843/

**Comment by ravipesala** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3078/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1847/

**Comment by QiangCai** [comment]
verified CI  successfully on local machine

**Pull request closed without merging**